api/app/Models/PoolCandidate.php

```diff
@@ -834,4 +839,112 @@ public function setApplicationSnapshot()
         $this->profile_snapshot = $profile;
         $this->save();
     }
+
+    public function computeDecisionsPerAssessmentStep()
+    {
+        $decisions = [];
+        $this->load([
+            'pool.assessmentSteps',
+            'assessmentResults',
+        ]);
+
+        foreach ($this->pool->assessmentSteps as $step) {
+            $stepId = $step->id;
+            $hasFailure = false;
+            $hasOnHold = false;
+            $hasToAssess = false;
+
+            $stepResults = $this->assessmentResults()->where('assessment_step_id', $stepId);
+            if (! $stepResults->count()) {
+                $decisions[$stepId] = null;
+
+                continue;
+            }
+
+            // Check assessed essential skills on this step
+            $essentialSkillAssessments = $stepResults->whereHas('assessmentStep', function ($query) {
+                $query->whereHas('poolSkills', function ($query) {
+                    $query->where('type', PoolSkillType::ESSENTIAL->name);
+                });
+            });
+
+            if (! $essentialSkillAssessments) {
+                $decisions[$stepId] = null;
+
+                continue;
+            }
+
+            foreach ($essentialSkillAssessments as $essentialSkillAssessment) {
+                $decision = $essentialSkillAssessment->assessment_decision;
+                if (! $decision) {
```

Comment:
Can this not be `case null` in the switch statement? (legitimate question - I'm not sure)

Reply:
`null` can, I need to check on `undefined`. Or is it impossible for this to be `undefined`? :thinking:

Reply:
I don't think PHP distinguishes between `null` and `undefined`, that's a javascript quirk 😆

Reply:
I have been stuck in TS land too long :stuck_out_tongue: Should be fixed in [move null check to switch](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10642/commits/028f765344ac96c6b24be2168138a7ef894f977b)